### PR TITLE
WIP: provide archive.org user id and ocaid in ia_auth response

### PR DESCRIPTION
**Pull request by hornc**
@rchrd2  Here are the changes we discussed, I'm still finding a way to test this does what we expect locally.

@mekarpeles for your review -- @rchrd2 and I were discussing the recent archive.org lending improvements and what needed to change on OL. This (hopefully!) provides the data archive.org needs to manage multiple loans and assign items to the correct user.

### CODE REVIEW

**Comment by rchrd2** [comment]
This looks good to me. I think in theory this would work. Is there a way for us to test on dev.openlibrary, before merging?

**Comment by hornc** [comment]
Definitely, we can push the branch to dev. @mekarpeles is best placed to do this as he has done it before. I only recently got deploy access and need hand holding ;)

**Comment by mekarpeles** [comment]
@hornc happy to help you through all of this on Monday (I'll feel so much less lonely on ol-dev0! Thank you!)

dev.openlibrary.org has the latest code from master w/ this branch played on top of it as `testing/lending-changes-cdl2`

File: openlibrary/plugins/upstream/borrow.py [hunk 1]
```diff
@@ -899,7 +899,7 @@ def get_ia_auth_dict(user, item_id, user_specified_loan_key, access_token):
         if not user_has_current_loan:
             raise Exception('lending: no current loan for this user found but no error condition specified')
 
-    return { 'success': True, 'token': make_ia_token(item_id, bookreader_auth_seconds) }
+    return { 'success': True, 'token': make_ia_token(item_id, bookreader_auth_seconds), 'userid': user.itemname(), 'ocaid': item_id }
```
Comment (mekarpeles) [inline openlibrary/plugins/upstream/borrow.py]: :+1:

**Comment by mekarpeles** [comment]
I have a followup patch for this running on dev, moving it to `WIP` until feedback from @rchrd2

**Comment by mekarpeles** [comment]
Closing in favor of #1354

**Pull request closed without merging**